src/django/api/views.py

```diff
@@ -3420,3 +3417,9 @@ def reject(self, request, pk=None):
                         facility_activity_report).data
 
         return Response(response_data)
+
+    def list(self, request):
```

Comment:
I've updated to place the admins-only restriction directly on the endpoint in cea5ba6.

Reply:
I would have thought this would work but when I used an API key associated with a user who was not staff, I was still able to get from the list endpoint.

After some searching I found this [StackOverflow answer](https://stackoverflow.com/a/39392875). It does not look like we can decorate the `list` function directly, but we can relatively easily create a custom subclass and assign that at the top of our ViewSet.

```diff
diff --git a/src/django/api/views.py b/src/django/api/views.py
index 503f643..85b0270 100644
--- a/src/django/api/views.py
+++ b/src/django/api/views.py
@@ -3348,6 +3348,15 @@ def update_facility_activity_report_status(facility_activity_report,
     return facility_activity_report
 
 
+class IsListAndAdminOrNotList(IsAdminUser):
+    """
+    Custom permission to only allow access to lists for admins
+    """
+    def has_permission(self, request, view):
+        is_admin = super(IsListAndAdminOrNotList, self).has_permission(request, view)
+        return view.action != 'list' or is_admin
+
+
 @schema(FacilityActivityReportAutoSchema())
 class FacilityActivityReportViewSet(viewsets.GenericViewSet):
     """
@@ -3355,6 +3364,7 @@ class FacilityActivityReportViewSet(viewsets.GenericViewSet):
     """
     queryset = FacilityActivityReport.objects.all()
     serializer_class = FacilityActivityReportSerializer
+    permission_classes = (IsListAndAdminOrNotList,)
 
     @action(detail=True, methods=['POST'],
             permission_classes=(IsAdminUser,),
@@ -3417,7 +3427,6 @@ class FacilityActivityReportViewSet(viewsets.GenericViewSet):
 
         return Response(response_data)
 
-    @permission_classes((IsAdminUser, ))
     def list(self, request):
         response_data = FacilityActivityReportSerializer(
             FacilityActivityReport.objects.all(), many=True).data
```

I may have not configured things correctly in my initial testing so please double-check that this change has an effect.

Reply:
Updating it to this and testing with `c5@example.com` is returning unauthorized appropriately for me.